dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/task/AbstractJdbcTask.java

```diff
@@ -111,29 +111,38 @@ protected CSVFormat newCsvFormat(@Nonnull ResultSet rs) throws SQLException {
 
   @Nonnull
   public ResultSetExtractor<Summary> newCsvResultSetExtractor(@Nonnull ByteSink sink) {
-    return rs -> {
-      try (RecordProgressMonitor monitor = new RecordProgressMonitor(getName())) {
-        printAllResults(sink, rs, monitor);
-        return new Summary(monitor.getCount());
-      } catch (IOException e) {
-        throw new SQLException(e);
-      }
+    return resultSet -> {
+      RecordProgressMonitor monitor = new RecordProgressMonitor(getName());
+      return printWithMonitor(sink, monitor, resultSet);
     };
   }
 
   @Nonnull
   protected ResultSetExtractor<Summary> newCsvResultSetExtractor(
       @Nonnull ByteSink sink, long count) {
-    return rs -> {
-      try (RecordProgressMonitor monitor = new RecordProgressMonitor(getName(), count)) {
-        printAllResults(sink, rs, monitor);
-        return new Summary(monitor.getCount());
-      } catch (IOException e) {
-        throw new SQLException(e);
-      }
+    return resultSet -> {
+      RecordProgressMonitor monitor = new RecordProgressMonitor(getName(), count);
+      return printWithMonitor(sink, monitor, resultSet);
     };
   }
 
+  private Summary printWithMonitor(
+      @Nonnull ByteSink sink,
+      @Nonnull RecordProgressMonitor monitor,
+      @CheckForNull ResultSet resultSet)
+      throws SQLException {
+    try {
+      if (resultSet != null) {
+        printAllResults(sink, resultSet, monitor);
+      }
+      return new Summary(monitor.getCount());
+    } catch (IOException e) {
+      throw new SQLException(e);
+    } finally {
+      monitor.close();
```

Comment:
I don't see the benefit of not using `monitor` in the `try-with-resources` statement. I don't think there needs to be an external resource to justify the usage of `try-with-resources`. It can be considered a syntax sugar for calling `close` in `finally` - I liked the previous version more.

As for the other two issues that are being addressed with the PR:
- the null check can be added directly inside `printAllResults`,
- the exception handling can also be moved to the inside of `printAllResults`.

Reply:
As discussed:
- I removed the `finally` block, reverting to try-with-resources
- the new method can stay, it decreases the complexity of `printAllResults` code